### DISCUSSION TITLE
Data Viewer discrepancies with CSVs

### DISCUSSION THREAD

**Discussion by cdc-jtmiller** [posit-dev/positron#5699] [Q&A] [answered]
Positron Version: 2024.12.0 (user setup) build 96
Code - OSS Version: 1.93.0
Commit: c5ce275dc502f6b15433b271802cb33e1ba5ef68
Date: 2024-11-28T02:53:40.010Z
Electron: 30.4.0
Chromium: 124.0.6367.243
Node.js: 20.15.1
V8: 12.4.254.20-electron.0
OS: Windows_NT x64 10.0.19045

When I have CSVs in a project folder, I can see the tabular representation of the data for some, but not all of the files.

In the first image, I have a CSV file that was created outside of Positron. In the second image, I'm using a CSV that is a subset of the first CSV. Positron recognizes the headers, but doesn't show the data.

Here are things I've tried:
Taking the original CSV (data shows fine), I've just removed rows / columns, then saved using both Excel and Notepad++. The resulting file doesn't show the tabular representation of the data.

Additionally, I've created a CSV, which again shows that Positron is reading the data, but doesn't show the data in tabular form.

I have tried (1) removing all extensions, (2) reverting my 'settings.json' file back to default and neither fixes the issue of showing all CSVs with their tabular representation.  I have (3) analyzed the CSV files and don't find any structural anomalies.

I have also created a CSV in Python, which also exhibits the lack of tabular representation in the data viewer.  (3rd image)

I don't know if this is a bug with Positron, or a problem with my settings. I have deleted Positron and re-installed. This did not fix the issue.  Any help or suggestion is appreciated.

Here is the code that created the CSV for image 3.

```
import pandas as pd
import numpy as np
import random
import string

# Set the number of rows and columns
num_rows = 5
num_columns = 5

# Generate random data
data = np.random.rand(num_rows, num_columns)

# Round the random numbers to two decimal places
data = np.round(data, 2)

# Create a DataFrame
df = pd.DataFrame(data, columns=[f'Column_{i+1}' for i in range(num_columns)])

# Add a few columns with random letters
num_text_columns = 5  # Number of text columns to add
for i in range(num_text_columns):
    df[f'Text_Column_{i+1}'] = [''.join(random.choices(string.ascii_letters, k=5)) for _ in range(num_rows)]

# Define the output file path
output_file_path = 'output/test_dat.csv'

# Export the DataFrame to a CSV file
df.to_csv(output_file_path, index=False, mode='w')
print(f"\nResults saved to: {output_file_path}")

# Created/Modified files during execution:
print(output_file_path)
```
![positron_csv1](https://github.com/user-attachments/assets/4cd567e6-c3e6-4421-9836-ffa0478486ed)
![positron_csv2](https://github.com/user-attachments/assets/7b0bdef6-4c6f-4f45-9ede-6d0c1e9d7eea)
![test_csv](https://github.com/user-attachments/assets/14aa6626-eca2-4359-8770-5aad4b65a4bb)

**Comment by jennybc** [comment, marked as answer]
This sounds a lot like #5552, in which we discovered that a csv with fewer than 10 rows is not viewable. That has since been fixed, but the fix is not yet in a release that's publicly available. I note that you are a couple of releases behind, so I suggest you update Positron soon-ish and I suspect you'll find that this has been fixed.

**Reply by cdc-jtmiller** [reply]
Thank you for the reply.  I think this must be the problem. I bumped up the number of rows in the sample program to 25 and the tabular data showed up.

I thought I had the latest version since I recently downloaded it, but have since located and updated to ver. 2025.01.0-39.  Even after installing, the old version was still showing, so I had to completely remove the older version and install the new one.

After a fresh install, the problem has gone away.

Thanks for you help. This issue was about to drive me crazy.

**Reply by jennybc** [reply]
Ah, yes [2025.01.0-39](https://github.com/posit-dev/positron/releases/tag/2025.01.0-39) should have the fix. Glad this is sorted.
